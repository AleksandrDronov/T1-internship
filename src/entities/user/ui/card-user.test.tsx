import renderer from "react-test-renderer";
import { CardUser } from ".";

describe("CardUser", () => {
  test("renders correctly", () => {
    const tree = renderer
      .create(<CardUser firstName="John" lastName="Doe" image="avatar.jpg" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("renders correctly with variant", () => {
    const tree = renderer
      .create(
        <CardUser
          variant="secondary"
          firstName="John"
          lastName="Doe"
          image="avatar.jpg"
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("renders correctly with isLoading", () => {
    const tree = renderer
      .create(
        <CardUser
          isLoading
          firstName="John"
          lastName="Doe"
          image="avatar.jpg"
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("renders correctly with isError", () => {
    const tree = renderer
      .create(
        <CardUser isError firstName="John" lastName="Doe" image="avatar.jpg" />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
