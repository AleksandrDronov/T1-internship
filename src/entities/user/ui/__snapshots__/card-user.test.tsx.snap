// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardUser renders correctly 1`] = `
<div
  aria-label="User card"
  className="container-primary"
>
  <img
    alt="Avatar of John Doe"
    className="image-primary"
    src="avatar.jpg"
  />
  <div>
    <p
      className="text b17-web gray-medium left"
    >
      Written By
    </p>
    <p
      className="text b18-web black left"
    >
      John
       
      Doe
    </p>
  </div>
</div>
`;

exports[`CardUser renders correctly with isError 1`] = `
<p
  className="title text b18-web black left"
>
  Error! Failed to load user
</p>
`;

exports[`CardUser renders correctly with isLoading 1`] = `
<p
  className="title text b18-web black left"
>
  Loading...
</p>
`;

exports[`CardUser renders correctly with variant 1`] = `
<div
  aria-label="User card"
  className="container-secondary"
>
  <img
    alt="Avatar of John Doe"
    className="image-secondary"
    src="avatar.jpg"
  />
  <div>
    <p
      className="text b20-web gray-medium left"
    >
      Written By
    </p>
    <p
      className="text b21-web black left"
    >
      John
       
      Doe
    </p>
  </div>
</div>
`;
